cmd/sonobuoy/app/results.go

```diff
@@ -254,6 +327,75 @@ func printResultsDetails(treePath []string, o *results.Item, input resultsInput)
 	return nil
 }
 
+// printClusterHealthResultsSummary prints the summary of the "fake" plugin for health summary,
+// tryingf to emulate the format of printResultsSummary
+func printClusterHealthResultsSummary(summary discovery.ClusterSummary) error {
+	fmt.Println("Run Details:")
+	fmt.Printf("API Server version: %s\n", summary.APIVersion)
+
+	fmt.Printf("Node health: %d/%d", summary.NodeHealth.Healthy, summary.NodeHealth.Total)
+	//Print the percentage only if Total is not 0 to avoid division by zero errors
+	if summary.NodeHealth.Total != 0 {
+		fmt.Printf(" (%d%%)", 100 * summary.NodeHealth.Healthy / summary.NodeHealth.Total)
+	}
+	fmt.Println()
+	//Details of the failed pods. Checking the slice lenght to avoid trusting the Total
+	if len(summary.NodeHealth.Details) > 0 && summary.NodeHealth.Healthy < summary.NodeHealth.Total {
+		fmt.Println("Details for failed nodes:")
+		for _, node := range summary.NodeHealth.Details {
+			if !node.Healthy {
+				fmt.Printf("%s Ready:%s: %s: %s\n", node.Name, node.Ready, node.Reason, node.Message)
+			}
+		}
+		fmt.Println()
+	}
+
+	//It might be nice to group pods by namespace.
+	//Also here, use len instead of trusting Total
+	if len(summary.PodHealth.Details) > 0 {
+		fmt.Printf("Pods health: %d/%d", summary.PodHealth.Healthy, summary.PodHealth.Total)
+		//Print the percentage only if Total is not 0 to avoid division by zero errors
+		if summary.PodHealth.Total != 0 {
+			fmt.Printf(" (%d%%)", 100 * summary.PodHealth.Healthy / summary.PodHealth.Total)
+		}
+		fmt.Println()
+		if summary.PodHealth.Healthy < summary.PodHealth.Total {
+			pods := make(map[string][]*discovery.HealthInfoDetails)
+			fmt.Println("Details for failed pods:")
+			//First, scan all the pods and extract the not ready ones, and group them by namespace
+			for podIdx, pod := range summary.PodHealth.Details {
+				if !pod.Healthy {
+					if _, ok := pods[pod.Namespace]; !ok {
+						pods[pod.Namespace] = make([]*discovery.HealthInfoDetails,0)
+					}
+					// Using a pointer so we don't need to make a copy
+					pods[pod.Namespace] = append(pods[pod.Namespace], &summary.PodHealth.Details[podIdx])
+				}
+			}
+			//And then print them, sorted by namespace
+			for namespace := range pods {
+				fmt.Println("Namespace: ", namespace)
+				for _, pod := range pods[namespace] {
+					fmt.Printf("%s/%s Ready:%s: %s: %s\n", pod.Namespace, pod.Name, pod.Ready, pod.Reason, pod.Message)
+				}
+				fmt.Println()
+			}
+		}
+	}
+
+	if len(summary.ErrorInfo) > 0 {
```

Comment:
Yes, review the log output too, I merged the commit because its changes are overlapping a bit for the ClusterSummary struct and the printClusterHealthResultsSummary, and the summary test, but I kept it a separate commit, so it's easy to back it out if necessary.